src/index.ts

```diff
@@ -1,7 +1,108 @@
+    }
 
-const initialize = (str: string) => {
-  console.log(str)
-};
+    // Get the royalties of a collection
+    async getCollectionRoyalties() {
+        const collectionRoyalties = await tokiumAPI({
+            method: 'POST',
+            url: '/getRoyalties',
+            data:{
+                collectionLink: this.collectionURL,
+            }
+        }).then((res) => {
+            return res.data;
+        }).catch((err) => {
+            console.log(err)
+        });
+        return collectionRoyalties
+    }
 
-export default initialize;
\ No newline at end of file
+    // Checks whether a wallet has NFT from a collection and returns data object
+    async hasNFT() {
+        const hasNFT = await tokiumAPI({
+            method: 'POST',
+            url: '/hasNFT',
+            data:{
+                collectionLink: this.collectionURL,
+                address: this.walletAddress,
+            }
+        }).then((res) => {
+            return res.data;
+        }).catch((err) => {
+            console.log(err)
+        });
+        return hasNFT
+    }
+
+    // Gets data from previous NFT transfers
+    async previousNftTransfers(mintAddress: string) {
+        if (!mintAddress) {
+            throw new Error('Token mint address is required to call the previousNftTransfer endpoint!')
+        }
+        const previousNftTransfers = await tokiumAPI({
+            method: 'POST',
+            url: '/previousNftTransfers',
+            data:{ tokenMintAddress: mintAddress},
+        }).then((res) => {
+            return res.data;
+        }).catch((err) => {
+            console.log(err)
+        });
+        return previousNftTransfers
+    }
+
+    // Gets the data from the last transfer
+    async lastTransfer(mintAddress: string) {
+        if (!mintAddress) {
+            throw new Error('Token mint address is required to call the lastTransfer endpoint!')
+        }
+        const lastTransfer = await tokiumAPI({
+            method: 'POST',
+            url: '/lastTransfer',
+            data:{ tokenMintAddress: mintAddress},
+        }).then((res) => {
+            return res.data;
+        }).catch((err) => {
+            console.log(err)
+        });
+        return lastTransfer
+    }
+
+    // Returns royalties paid and tokens owned 
+    async hasPaidRoyalties() {
+        const verified = await tokiumAPI({
+            method: 'POST',
+            url: '/hasPaidRoyalties',
+            data: {
+                    collectionLink: this.collectionURL,
+                    address: this.walletAddress,
+                }
+        }).then((res) => {
+            return res.data;
+        }).catch((err) => {
+            console.log(err)
+        });
+        return verified
+    }
+
+    // Returns boolean for if validated
+    async verifyTokenWithRoyalty() {
+        const { royaltiesPaid, tokensOwned} = await this.hasPaidRoyalties();
+        if (royaltiesPaid === true && tokensOwned > 0) {
```

Comment:
if royalties paid returns true, the tokensOwned will always be > 0. So the second check could be removed